.pre-commit-config.yaml

```diff
@@ -0,0 +1,37 @@
+exclude: "docs|node_modules|migrations|.git|.tox"
+default_stages: [commit]
+fail_fast: true
+
+repos:
+    - repo: https://github.com/pre-commit/pre-commit-hooks
+      rev: v3.4.0
+      hooks:
+          - id: trailing-whitespace
+          - id: end-of-file-fixer
+          - id: check-yaml
+
+    - repo: https://github.com/psf/black
+      rev: 21.9b0
+      hooks:
+          - id: black
+
+    - repo: https://github.com/timothycrosley/isort
+      rev: 5.9.3
+      hooks:
+          - id: isort
+
+    # Need to use flake8 GitHub mirror due to CentOS git issue with GitLab
+    # https://github.com/pre-commit/pre-commit/issues/1206
+    - repo: https://github.com/pycqa/flake8
+      rev: 3.9.2
+      hooks:
+          - id: flake8
+            args: ["--config=setup.cfg"]
+            additional_dependencies: [flake8-isort]
+            exclude: ^sample_setups/jsons/
+
+    - repo: https://github.com/pre-commit/mirrors-mypy
+      rev: v0.910
+      hooks:
+          - id: mypy
+            args: ["--config=setup.cfg"]
```

Comment:
Same comment regarding running the tools and fixing issues.

`mypy` is an optional static type checker. It may catch some existing issues that need to be fixed.